STEP_Project/OUSelectContent.js

```diff
@@ -19,14 +25,31 @@ function selectClick ()
             if (data[dataid] != undefined)
             {
                 data[dataid] = dataname;
-                chrome.storage.sync.set({[dataid]: dataname});
+                storageObj.set({[dataid]: dataname});
             }
         }
     });
 }
 
+/**
+ * This functions adds a message listener to this content script
+ * Received message is expected to be the data id from preferred.js
+ * It then performs a "click" on the required OU button.
+ */
+function applySelect()
+{
+  chrome.runtime.onMessage.addListener(
+    function(request) 
+    {
+      let queryVar = "[data-content-id='" + request.dataId + "']";
+      let OU = document.querySelectorAll(queryVar);
+      OU[0].children[1].children[0].click();
```

Comment:
I think we should mock/fake chrome.runtime.onMessage and test this piece of code as well. We can manually trigger the callback function from the faked object. You can also check for element click, check out https://stackoverflow.com/questions/23486444/error-on-jasmine-expected-event-click-to-have-been-triggered-on-div-id/23492830#23492830.

Reply:
Added additional testcase by mocking _chrome.runtime.onMessage_ and the _request_ object received as a message.